packages/ui/src/components/Table/STable.vue

```diff
@@ -0,0 +1,819 @@
+const emit = defineEmits<{
+  (event: 'cell-mouse-enter', ...value: CellEventData): void
+  (event: 'cell-mouse-leave', ...value: CellEventData): void
+  (event: 'cell-click', ...value: CellEventData): void
+  (event: 'cell-dblclick', ...value: CellEventData): void
+  (event: 'header-click', ...value: HeaderEventData): void
+  (event: 'header-contextmenu', ...value: HeaderEventData): void
+  (event: 'row-click', ...value: RowEventData): void
+  (event: 'row-dblclick', ...value: RowEventData): void
+  (event: 'row-contextmenu', ...value: RowEventData): void
+  (event: 'sort-change', value: SortEventData): void
+  (event: 'selection-change', value: TableRow[]): void
+  (event: 'select-all', value: TableRow[]): void
+  (event: 'select', ...value: [TableRow[], TableRow]): void
+  (event: 'expand-change', ...value: [TableRow, TableRow[]]): void
+  (event: 'current-change', ...value: [TableRow | null, TableRow | null]): void
+}>()
+
+const columns: (ColumnApi | ActionColumnApi)[] = shallowReactive([])
+const { data, selectOnIndeterminate, fit, showHeader, height, maxHeight, expandRowKeys, currentRowKey } = toRefs(props)
+const rowKeys = shallowReactive(new Map<TableRow, unknown>())
+const keyRows = shallowReactive(new Map<unknown, TableRow>())
+const activeExpandColumn = computed(() => findLast(columns, isExpandColumn))
+const activeSelectionColumn = computed(() => findLast(columns, isSelectionColumn))
+
+const tableWrapper: MaybeElementRef = ref(null)
+const tableSizes = reactive({
+  height: 0,
+  width: 0,
+})
+useResizeObserver(tableWrapper, (entries) => {
+  tableSizes.width = entries[0].contentRect.width
+  tableSizes.height = entries[0].contentRect.height
+})
+
+const headerWrapper: MaybeElementRef = ref(null)
+const headerHeight = ref(0)
+useResizeObserver(headerWrapper, (entries) => {
+  headerHeight.value = entries[0].contentRect.height
+})
+whenever(not(showHeader), () => {
+  headerHeight.value = 0
+})
+
+const { columnsWidths, columnsWidthsSum } = useFlexColumns(columns, toRef(tableSizes, 'width'), fit)
+const { tableHeightStyles, bodyHeightStyles } = useTableHeights({
+  propHeight: height,
+  propMaxHeight: maxHeight,
+  headerHeight,
+  tableHeight: toRef(tableSizes, 'height'),
+})
+const { expandedRows, toggleRowExpanded } = useColumnExpand()
+const { sortState, sortedData, sortExplicitly, handleSortChange, getNextOrder, applyCurrentSort, clearSort } =
+  useColumnSort(data)
+const { selectedRows, isAllSelected, isSomeSelected, toggleAllSelection, toggleRowSelection } = useRowSelect(
+  sortedData,
+  reactive({ selectOnIndeterminate }),
+)
+
+const currentRow: ShallowRef<TableRow | null> = shallowRef(null)
+
+function isCurrentRow(row: TableRow) {
+  return currentRow.value === toRaw(row)
+}
+
+function setCurrentRow(row: TableRow | null) {
+  currentRow.value = toRaw(row)
+}
+
+watch([keyRows, currentRowKey], () => {
+  setCurrentRow(keyRows.get(currentRowKey.value) ?? null)
+})
```

Comment:
Shouldn't it fire immediately?

Reply:
`keyRows` initializes first, and triggers this callback